### PR TITLE
Update NEWS file for new alpha release

### PR DESCRIPTION
**Pull request by mattcaswell**
Updated the NEWS file with the most significant items from CHANGES

I think this should go in before the alpha1 release

### CODE REVIEW

**Comment by levitte** [comment]
> In these days of public git logs, is it time to retire NEWS or CHANGES?

Absolutely not.  (Some) people who download our tarballs do read them, *especially since we urge them to* (for example for information on type becoming opaque).

**Comment by mattcaswell** [comment]
> In these days of public git logs, is it time to retire NEWS or CHANGES?

I think they do provide a useful source of documentation. Reading through the git log would be a very difficult way of figuring out the main changes. The CHANGES file is relatively easy to read and provides that summary view. Similarly the NEWS file is more of the "executive summary" ;-)

**Comment by richsalz** [comment]
I am not sure we really need *both* files.  Certainly one of them.

**Comment by mattcaswell** [comment]
BTW, I just realised that there doesn't seem to be any mention in CHANGES of the RAND/DRBG changes that were made. That seems fairly significant and worthy of an entry? @mspncp

**Comment by richsalz** [comment]
This brings up a problem with the current system.  How do we know when to put something in CHANGES?  I an assume that NEWS turns each CHANGES entry into a one-liner, but the first issue?

**Comment by mattcaswell** [comment]
> I an assume that NEWS turns each CHANGES entry into a one-liner, but the first issue?

I haven't put everything from CHANGES into NEWS in this PR. As I said in the description of this PR I just read through it and pulled out the things that I thought were the most significant

**Comment by levitte** [comment]
NEWS has usually been the broad user-visible changes, such as added features, removed features or changes that affect users in a big way (structures becoming opaque).  CHANGES has been more details, bigger descriptions of changes and also described bigger internal changes that don't directly affect the user

So no, CHANGES and NEWS aren't meant to have some kind of 1:1 mapping, that's entirely false.  However, we've been reaaaaally sloppy with CHANGES lately.

There's still a difference between CHANGES and the git log.  We don't put in a note on every typo in CHANGES, nor do we note code that's been reverted, or every itty gritty bug correction.  And most of all, we can make "history changes" in CHANGES, while we cannot do so with the git log, so while CHANGES is supposed to be a log, it's still supposed to be a reader's digest compared to the git log.

**Comment by mspncp** [comment]
> BTW, I just realised that there doesn't seem to be any mention in CHANGES of the RAND/DRBG changes that were made. That seems fairly significant and worthy of an entry? @mspncp

I won't manage to do it in time for the freeze today, but I intend to add _a lot of documention_ for the new RAND_DRBG api in the next weeks, hopefully until the next alpha. This will include a few lines to the changelog, of course.

**Comment by mattcaswell** [comment]
> I won't manage to do it in time for the freeze today, but I intend to add a lot of documention for the new RAND_DRBG api in the next weeks, hopefully until the next alpha. This will include a few lines to the changelog, of course.

Great (BTW we're already frozen, but I would have made an exception for a CHANGES update, since this is very low risk).

File: NEWS [hunk 1]
```diff
@@ -12,6 +12,14 @@
...
       o Add a STORE module (OSSL_STORE)
       o Claim the namespaces OSSL and OPENSSL, represented as symbol prefixes
+      o Add multi-prime RSA (RFC 8017) support
+      o Add SM3 implemented according to GB/T 32905-2016
+      o Add SM4 implemented according to GB/T 32907-2016.
+      o Add 'Maximum Fragment Length' TLS extension negotiation and support
+      o Add ARIA support
+      o Add SHA3
+      o Rewrite of devcrypto engine
+      o Add support for SipHash
```
Comment (mspncp) [inline NEWS]: @mattcaswell, if it's not too late yet, here is a raw draft for NEWS about the RNG. Probably it's too verbose, and probably @richsalz should have a look over it.

- Replace the default RAND method by new implementation using a CTR-DRBG (Deterministic Random Bit Generator based on AES-CTR) according to NIST SP800-90Ar1. It is essentially a port of the default RAND method from the FIPS 2.0 Object Module.
- Keep the DRBG state in secure storage, if it is enabled.
- Added seed chaining of DRBG instances. Every SSL object now has its own DRBG instance, to reduce lock contention and increase unpredictability of random output.
- Use a dedicated DRBG for generating long term private keys.


_TODO:_ Is there already an entry about secure storage? I didn't see anything.

Comment (mspncp) [reply NEWS]: - The DRBG supports automatic seeding and reseeding without depending on support from the applicaion.

**Comment by mattcaswell** [comment]
@mspncp, if you can get the updates done in the form of a PR to update CHANGES and NEWS then I can consider it before I do the release.

**Comment by mattcaswell** [comment]
This PR is pushed. Closing.

**Pull request closed without merging**